main/src/main/java/cgeo/geocaching/unifiedmap/layers/ElevationChartUtils.java

```diff
@@ -0,0 +1,153 @@
+            dataSet.setColor(color);
+            dataSet.setHighLightColor(color);
+            dataSet.setValueTextColor(res.getColor(R.color.colorText));
+            dataSet.setDrawValues(false);
+            dataSet.setDrawFilled(true);
+            dataSet.setFillColor(color);
+            dataSet.setDrawCircles(false);
+
+            final LineData lineData = new LineData(dataSet);
+            chart.setData(lineData);
+
+            chart.setExtraOffsets(0, 0, 0, 20);
+            chart.setBackgroundColor(res.getColor(R.color.colorBackground));
+
+            final Legend legend = chart.getLegend();
+            legend.setEnabled(false);
+            chart.getDescription().setEnabled(false);
+
+            final XAxis xAxis = chart.getXAxis();
+            xAxis.setPosition(XAxis.XAxisPosition.BOTTOM);
+            xAxis.setDrawAxisLine(true);
+            xAxis.setTextSize(getDimensionInDp(res, R.dimen.textSize_detailsSecondary));
+            xAxis.setTextColor(res.getColor(R.color.colorText));
+            xAxis.setAxisMinimum(0.0f);
+            xAxis.setValueFormatter((value, axis) -> Units.getDistanceFromKilometers(value));
+
+            final YAxis yAxis = chart.getAxisLeft();
+            yAxis.setDrawAxisLine(true);
+            yAxis.setTextSize(getDimensionInDp(res, R.dimen.textSize_detailsSecondary));
+            yAxis.setTextColor(res.getColor(R.color.colorText));
+
+            final YAxis yAxis2 = chart.getAxisRight();
+            yAxis2.setEnabled(false);
+
+            // follow tap on elevation chart in route on map
+            chart.setOnChartValueSelectedListener(new OnChartValueSelectedListener() {
+                @Override
+                public void onValueSelected(final Entry e, final Highlight h) {
+                    // find position in route corresponding to distance from start as given in e.getX()
+                    float distance = 0.0f;
+                    Geopoint lastPoint = null;
+                    Geopoint center = null;
+                    for (RouteSegment segment : route.getSegments()) {
+                        for (Geopoint point : segment.getPoints()) {
+                            if (lastPoint != null) {
+                                distance += lastPoint.distanceTo(point);
+                            }
+                            if (distance >= e.getX()) {
+                                center = point;
+                                break;
+                            }
+                            lastPoint = point;
+                        }
+                    }
+                    // update marker if position found
+                    if (center != null) {
+// icon marker isn't drawn
+                        final GeoItem marker = GeoPrimitive.createMarker(center, GeoIcon.builder().setBitmap(ImageUtils.convertToBitmap(ResourcesCompat.getDrawable(CgeoApplication.getInstance().getResources(), R.drawable.circle, null))).build());
```

Comment:
You may want to add `.buildUpon().setZLevel(LayerHelper.ZINDEX_XYZ).build()` to specify a ZLevel. Don't know whether this makes a difference. At least I used it nearly everywhere.

You can also take a look at `CoordsIndicatorLayer` which does pretty much the same as you need here.

Reply:
I've added ZLayer anyway to keep the marker on top